dataflowengineoss/src/main/scala/io/joern/dataflowengineoss/slicing/DataFlowSlicing.scala

```diff
@@ -13,30 +15,42 @@ object DataFlowSlicing {
   def calculateDataFlowSlice(cpg: Cpg, config: DataFlowConfig): Option[DataFlowSlice] = {
     implicit val implicitConfig: BaseConfig = config
 
+    val exec = config.parallelism match
+      case Some(parallelism) if parallelism == 1 => Executors.newSingleThreadExecutor()
+      case Some(parallelism) if parallelism > 1  => Executors.newWorkStealingPool(parallelism)
+      case _                                     => Executors.newWorkStealingPool()
+
     (config.fileFilter match {
-      case Some(fileName) => cpg.file.nameExact(fileName).ast.isCall
+      case Some(fileName) => cpg.file.nameExact(fileName).method.call
       case None           => cpg.call
-    }).iterator.method.withMethodNameFilter.withMethodParameterFilter.withMethodAnnotationFilter.ast.isCall
-      .flatMap { c =>
-        val sinks           = config.sinkPatternFilter.map(filter => c.argument.code(filter).l).getOrElse(c.argument.l)
-        val sliceNodes      = sinks.iterator.repeat(_.ddgIn)(_.maxDepth(config.sliceDepth).emit).dedup.l
-        val sliceNodesIdSet = sliceNodes.id.toSet
-        // Lazily set up the rest if the filters are satisfied
-        lazy val sliceEdges = sliceNodes
-          .flatMap(_.outE)
-          .filter(x => sliceNodesIdSet.contains(x.inNode().id()))
-          .map { e => SliceEdge(e.outNode().id(), e.inNode().id(), e.label()) }
-          .toSet
-        lazy val slice = Option(DataFlowSlice(sliceNodes.map(cfgNodeToSliceNode).toSet, sliceEdges))
+    }).iterator.method.withMethodNameFilter.withMethodParameterFilter.withMethodAnnotationFilter.call
```

Comment:
This filter is still slower than `where(c => c.callee.isExternal)`. In addition, we need to filter trivial stdlib calls such as `java.lang`

```
c.typeFullName.startsWith("java."))
```